models/schemas/threads.py

```diff
@@ -0,0 +1,166 @@
+# MIT License
+#
+# Copyright (c) 2020-2024 Send A Hug
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# The provided Software is separate from the idea behind its website. The Send A Hug
+# website and its underlying design and ideas are owned by Send A Hug group and
+# may not be sold, sub-licensed or distributed in any way. The Software itself may
+# be adapted for any purpose and used freely under the given conditions.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from models.base_models import BaseModel, DumpedModel
+from models.schemas.messages import Message
+from models.schemas.users import User
+
+
+from sqlalchemy import ForeignKey, Integer, and_, case, false, func, or_, select, true
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy.orm import Mapped, column_property, mapped_column, relationship
+
+
+import json
+
+
+class Thread(BaseModel):
```

Comment:
Same here, shouldn't this be in messaging? It's basically just a mapping table for that

Reply:
makes sense.